source/auth/auth.rst

```diff
@@ -1251,172 +1246,514 @@ mechanism
 
 mechanism_properties
     PROVIDER_NAME
-        Drivers MUST allow the user to specify a name for using a service
-        to obtain credentials that is one of ["aws"].
-    REQUEST_TOKEN_CALLBACK
-        Drivers MUST allow the user to specify a callback of the form
-        "onRequest" (defined below), if the driver supports
-        providing objects as mechanism property values.  Otherwise the driver MUST allow it as a MongoClientOption.
-    REFRESH_TOKEN_CALLBACK
-        Drivers MUST allow the user to specify a callback of the form
-        "onRefresh" (defined below), if the driver supports
-        providing objects as mechanism property values.  Otherwise the driver MUST allow it as a MongoClientOption.
+        The name of a built-in OIDC provider integration to use to obtain
+        credentials. The value MUST be one of ["aws"]. If both ``PROVIDER_NAME``
+        and an `OIDC Callback`_ are provided for the same ``MongoClient``
+        (either via the ``CALLBACK`` mechanism property or a ``MongoClient``
+        configuration), the driver MUST raise an error.
+
+    CALLBACK
+        An `OIDC Callback` that returns credentials for OIDC providers that do
+        not have a built-in integraiton. Drivers MAY allow the user to specify
+        an `OIDC Callback`_ using a ``MongoClient`` configuration instead of a
+        mechanism property, depending on what is conventional for the driver.
+        Drivers MUST NOT support both the ``CALLBACK`` mechanism property and
+        the ``MongoClient`` configuration.
+
+    CALLBACK_TYPE
+        The type of `OIDC Callback`. The value MUST be one of ["machine",
+        "human"]. If an `OIDC Callback`_ is configured and ``CALLBACK_TYPE`` is
+        not specified, the driver MUST raise an error. This property is only
+        required for drivers that support the `Human Authentication Flow`_.
```

Comment:
Yeah, that makes sense. It would be weird if they were specified in different places. I also added a note that drivers can use a `MongoClient` config to specify the callback type if they also use a `MongoClient` config to specify the callback.

Reply:
Based on Jeff's feedback, my plan is to actually remove `CALLBACK_TYPE` and replace it with either a separate callback argument specifically for the human callback or a boolean value to enable the human auth flow. In either case, I will still require that it not be specified in the connection string.